### PR TITLE
[docs] Improvements: Remove inconsistent use of emojis

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Refs ENG-5361

# How

This PR addresses the issue of removing inconsistent use of emojis across docs. Also, use `<YesIcon />` and `<NoIcon />` components, wherever necessary, to replace ✅ & ❌ 

Other improvements:
- fixes some internal links that were not working previously
- replaces legacy linking (that is, `../some-link.md` to `/some-link`)
- fixes some typos in "Conceptual Overview" pages

# Test Plan

# Checklist

All changes have been tested by running the `/docs` locally.

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/guides/config-plugins.md [hunk 1]
```diff
@@ -3,6 +3,7 @@ title: Config Plugins
 ---
 
 import { Terminal } from '~/ui/components/Snippet';
+import {YesIcon, NoIcon} from '~/ui/components/DocIcons';
```
Comment (jonsamp) [inline docs/pages/guides/config-plugins.md:6]:
```suggestion
import { YesIcon, NoIcon } from '~/ui/components/DocIcons';
```

**Review by jonsamp — APPROVED** [review]
Thank you for removing all of these. This is a big improvement!

File: docs/pages/expokit/expokit.md [hunk 2]
```diff
@@ -109,74 +109,77 @@ with
 
 If upgrading from SDK 32 or below:
```
Comment (Simek) [inline docs/pages/expokit/expokit.md:110]: It's probably out of this PR scope, but we should remove those instructions completely, since they are referring to the few years old releases.

File: docs/pages/guides/using-custom-fonts.md [hunk 3]
```diff
@@ -112,19 +113,19 @@ If you need to use another format, you may need to [customize the Metro bundler
...
-| woff   | ✅  | ✅  |         |
-| woff2  | ✅  | ✅  |         |
-| dfont  |     |     | ✅      |
-| svg    | ✳️  |     |         |
-| eot    | ✳️  |     |         |
-| fon    |     |     |         |
-| bdf    |     |     |         |
-| ps     |     |     |         |
-| ttc    |     |     |         |
+| Format | Web         | iOS         | Android     |
```
Comment (Simek) [inline docs/pages/guides/using-custom-fonts.md:116]: What do you think about filling empty cells with `NoIcon`?

Comment (amandeepmittal) [reply docs/pages/guides/using-custom-fonts.md:116]: I was wondering if that will be a good way to go. Thanks for the heads up. I'll make that change.

Comment (Simek) [reply docs/pages/guides/using-custom-fonts.md:116]: Also it would be nice to reorder the columns in the alphabetical order.

File: docs/pages/guides/using-nextjs.md [hunk 4]
```diff
@@ -147,12 +144,12 @@ This is Vercel's preferred method for deploying Next.js projects to production.
...
 ### Polyfill setImmediate
 
-> 💡 Fixes `setImmediate is not defined` error.
+> Fixes `setImmediate is not defined` error.
 
 A lot of libraries in the React ecosystem use the `setImmediate()` API (like `react-native-reanimated`), which Next.js doesn't polyfill by default. To fix this you can polyfill it yourself.
 
 - Install: `yarn add setimmediate`
-- Import in **pages/_app.js**, at the top of the file:
+- Import in **pages/\_app.js**, at the top of the file:
```
Comment (Simek) [inline docs/pages/guides/using-nextjs.md:152]: Hmm, was this change intentional?

Comment (amandeepmittal) [reply docs/pages/guides/using-nextjs.md:152]: Nope. It sneaked past me. I guess my VSCode config are broken after recently updating it. I'll revert this. Thanks for pointing this out 👍

File: docs/pages/expokit/expokit.md (hunk 2 above)
Comment (amandeepmittal) [reply docs/pages/expokit/expokit.md:110]: Do you mean delete the ExpoKit docs from the archive section?

**Review by Simek — APPROVED** [review]
Great work on those changes Aman! 🙏 

I have left few small suggestions and questions, but generally this changeset LGTM!

File: docs/pages/expokit/expokit.md (hunk 2 above)
Comment (Simek) [reply docs/pages/expokit/expokit.md:110]: Oh, good reminder, I have forgot that those pages are a part of archive/deprecated section. In this case we probably should leave those pages as-is. 🙂
